### ISSUE TITLE
Following up loading really long web page issue... (#85)

### ISSUE THREAD

**Issue by sanjeevApple** [gildas-lormeau/single-file-cli#86]
Couple of other problems after the fix in version 2.0.36

1) It seems like there is another issue, the fix works for Chrome but not for Deno browser, which is not a big deal, but wanted to let you know.

So this works:

./single-file https://www.apple.com/macbook-air macbook-air.html --browser-executable-path="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome" --browser-height=50000 --browser-wait-until="load" --load-deferred-images-keep-zoom-level=true

but this hangs when I remove the browser executable path and use Deno

./single-file https://www.apple.com/macbook-air macbook-air.html --browser-height=50000 --browser-wait-until="load" --load-deferred-images-keep-zoom-level=true

2) It is back to the original problem, for some reason the fix didn't help. Even with scrolling pixel height of roughly 30000 hangs with browser height of 30000 for airpods-max.

./single-file https://www.apple.com/airpods-max airpods-max.html --browser-executable-path="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome" --browser-height=30000 --browser-wait-until="load" --load-deferred-images-keep-zoom-level=true

hangs..

3) Also what I observed is that if you download the airpods-max page in a browser, physically scroll to the bottom, and use the plugin to save the file, there is no problem at all and also it is superfast, under 5 seconds, guess the page is already in the browser. But when you use single-file cli, there are a couple of problem, one is the hang, and second is the speed, takes a minute or so compared to 5 seconds. Wondering what is so different between browser extension vs cli, even though it is using the same chrome browser to render the web page and then saving it. The only other difference would be physical scrolling vs --browser-height=50000.

Thanks
Cheers...

**Comment by sanjeevApple** [comment]
I have Google Chrome installed, the Version is 124.0.6367.119 (Official Build) (arm64). So not quite sure why it hung when --browser-executable-path was not specified. It should have picked up Google Chrome, don't have Chromium installed. I can try again. May be the hang was for a different reason.

I just tried this:

./single-file https://www.apple.com/airpods-max airpods-max.html --browser-executable-path="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome" --browser-height=30000 --browser-wait-until="load"

and it just hangs... Is this working for you? I do have the latest build with version 2.0.36

./single-file --version
2.0.36

**Comment by sanjeevApple** [comment]
Wondering if there is something in your local environment which is not reflected in the build, may be some settings or the default options (we are only overriding browser height and browser wait until), which is still causing it to hang when run directly by downloading the latest release from the git...

**Comment by gildas-lormeau** [comment]
If confirm that the command you mentioned works fine for me. Maybe you should remove `--browser-wait-until="load"` or replace it with `--browser-wait-until="networkAlmostIdle"`.

**Comment by sanjeevApple** [comment]
In both cases with either --browser-wait-until property removed or changed to networkAlmostIdle, it is still stuck. When I Ctrl-C out of it, this is what I see...

**Comment by sanjeevApple** [comment]
Uncaught (in promise) NotFound: No such file or directory (os error 2): remove '/var/folders/g5/4hp_ppfs47g8gmd1w6xp0_cm0000gn/T/2aad9786888d493'
		await remove(profilePath, { recursive: true });
		^
    at async Object.remove (ext:deno_fs/30_fs.js:260:3)
    at async closeBrowser (file:///Users/sanjeev/Downloads/single-file-cli/lib/browser.js:120:3)
    at async closeBrowserAndExit (file:///Users/sanjeev/Downloads/single-file-cli/single-file-launcher.js:122:2)

**Comment by sanjeevApple** [comment]
the url apple.com works fine with different browser heights, 20K, 30K and 50K.

the apple.com/airpods-max, works fine with no height (lots of images are missing, so that won't work), at height 20K it take 6 min but is still missing a lot of images. At height 35K or 50K it just hangs, this is without browser wait delay option.

I tried it on another laptop, Apple M2 Pro, same behavior, it hangs at 35K, 50K height.

**Comment by Marc-Cox-08** [comment]
Thank you I really like Singlefile. So taking time to document the following: 
How to reproduce Singlefile stuck on deferred images for DEBUG
Step #1) go to this link :  https://www.cbsnews.com/colorado/news/winter-skipped-western-united-states/
Step #2) (Try) Save page with Singlefile, NOTE Singlefile stuck UNTIL Pause Video(s), 
ONLY AFTER videos are paused Singlefile will save file.

Google Ai suggests "Disable "Save Deferred Images":" ;
I think that Google Ai suggestion is overkill, and would prefer my solutions:

Quick Bug Fix: I suggest you set configurable timeout for Save Deferred Images to a default of 1-2 seconds.

Better/Harder/Sophisticated Fix: Do NOT wait on Deferred Images to complete, 
just take a snapshot image of ONLY that Deferred Images section of page/screen and save that.

HTH and let me know if you adopt any of those solutions, or have better recommendations.
